specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachineScaleSet.json

```diff
@@ -6336,17 +6340,24 @@
       "properties": {
         "id": {
           "type": "string",
-          "description": "The security posture reference id in the form of /CommunityGalleries/{communityGalleryName}/securityPostures/{securityPostureName}/versions/{major.minor.patch}|{major.*}|latest"
+          "description": "The security posture reference id in the form of /CommunityGalleries/{communityGalleryName}/securityPostures/{securityPostureName}/versions/{major.minor.patch}|latest"
         },
         "excludeExtensions": {
           "type": "array",
           "items": {
-            "$ref": "./virtualMachine.json#/definitions/VirtualMachineExtension"
+            "type": "string"
           },
-          "description": "List of virtual machine extensions to exclude when applying the Security Posture."
+          "description": "The list of virtual machine extension names to exclude when applying the security posture."
+        },
+        "isOverridable": {
```

Comment:
>isOverridable [](http://example.com/codeflow?start=9&length=13)

Replace boolean/switch properties with a more meaningful enum whenever possible. A boolean will forever have two valid values (true or false). Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].

For example, could this be overridable by other entities in the future other than user and may want to differentiate

Reply:
Appreciate the insight on this. I talked this over with my team and we're in agreement that the `boolean` is the best representation of the nature of the `isOverridable` property. This property will only ever have two states and moving to an `enum` would introduce new complexity and be antipattern to what we're trying to accomplish with the property.

Sharing at least one other instance in `Microsoft.Compute` where there is a boolean field using a similar naming convention for reference (e.g. `isVMInStandByPool`) - https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/instance-view?view=rest-compute-2024-03-01&tabs=HTTP#virtualmachineinstanceview